src/math/trig.c

```diff
@@ -43,6 +43,65 @@ const int32_t cord_arcsincos_q30fl  = Q_CONVERT_FLOAT(1.0000000000000000, 30);
 /**
  * \CORDIC-based approximation of sine, cosine and complex exponential
  */
+#ifdef CORDIC_HIFI3
+void cordic_approx(int32_t th_rad_fxp, int32_t a_idx, ae_int32 *sign, ae_int32 *b_yn,
+		   ae_int32 *xn, ae_int32 *th_cdc_fxp)
+{
+	ae_f32x2 xy_tmp;
+	ae_f32x2 yn_xn;
+	int32_t b_idx;
+	*sign = 1;
+
+	/* Addition or subtraction by a multiple of pi/2 is done in the data type
+	 * of the input. When the fraction length is 29, then the quantization error
+	 * introduced by the addition or subtraction of pi/2 is done with 29 bits of
+	 * precision.Input range of cordicsin must be in the range [-2*pi, 2*pi),
+	 * a signed type with fractionLength = wordLength-4 will fit this range
+	 * without overflow.Increase of fractionLength makes the addition or
+	 * subtraction of a multiple of pi/2 more precise
+	 */
+	if (th_rad_fxp > cord_sincos_piovertwo_q28fl) {
+		if ((th_rad_fxp - cord_sincos_piovertwo_q29fl) <= cord_sincos_piovertwo_q28fl) {
```

Comment:
This could be simplified to if (th_rad_fxp <= CONSTANT) where #define CONSTANT would be computed by cpp.